### PR TITLE
update docs for Infra projects 

### LINKED ISSUE #328: update docs for Infra projects
Shippable/pm#5907

### PR DESCRIPTION
**Pull request by ankul-shippable**
closes #328

### CODE REVIEW

File: sources/pipelines_configure.md [hunk 1]
```diff
@@ -113,8 +113,53 @@ To create an environment on Shippable, you should have installed the Universal C
 
 You have created your environment and you're now ready to start creating your pipelines.
 
-#### Using a config file in Terraform format
-Coming soon....
+### Provisioning infrastructure using Terraform
+You can provision infrastructure using terraform by following the steps below:
+
+* Create a terraform project with required infrastructure configurations. you can check our [sample terraform project](https://github.com/shippableSamples/sample_infra_terraform/) to get you started.
```
Comment (rageshkrishna) [inline sources/pipelines_configure.md]: Create a terraform project with **the** required infrastructure configuration~~s~~. You can check our [sample terraform project](https://github.com/shippableSamples/sample_infra_terraform/) to get ~~you~~ started.

File: sources/pipelines_configure.md [hunk 2]
```diff
@@ -113,8 +113,53 @@ To create an environment on Shippable, you should have installed the Universal C
...
+* On your subscriptions page goto `Infra` tab, on top-right corner click on `Enable Project`. A list of unenabled projects will come up, you can enable required project and enable it.
```
Comment (rageshkrishna) [inline sources/pipelines_configure.md]: On your subscriptions page go **space** to **the** `Infra` tab and click on `Enable Project` in the top-right corner. Find your terraform project in the list and click on the `Enable` button.

File: sources/pipelines_configure.md [hunk 3]
```diff
@@ -113,8 +113,53 @@ To create an environment on Shippable, you should have installed the Universal C
...
+* After enabling you project will show up in `Infra` tab. you can provision infrastructure by clicking on provision.
```
Comment (rageshkrishna) [inline sources/pipelines_configure.md]: Your project will now show up in the `Infra` tab. Click on the `Provision` button to start provisioning your infrastructure.

File: sources/pipelines_configure.md [hunk 4]
```diff
@@ -113,8 +113,53 @@ To create an environment on Shippable, you should have installed the Universal C
...
+  on_success:
+  on_failure:
+
+integrations:
+  deploy:
+  hub:
+  notifications:
+  key:
+```
+* If `prov` section of `infra` tag is empty, default command `terraform apply` will be executed.
```
Comment (rageshkrishna) [inline sources/pipelines_configure.md]: If **the** `prov` section of **the** `infra` tag is empty, `terraform apply` will be executed by default.

File: sources/pipelines_configure.md [hunk 5]
```diff
@@ -113,8 +113,53 @@ To create an environment on Shippable, you should have installed the Universal C
...
+* Your state files are saved at `/shippableci/shippable/provision` directory before provisioning, in terraform commands you can specify path to state files as shown in the above shippable.yml file. 
+* Please note that your shippable.yml file should not contain tags which adds to multiple job creation. Multiple jobs will result in provisioning of multiple infrastructures.
```
Comment (rageshkrishna) [inline sources/pipelines_configure.md]: Not sure what you mean by multiple jobs? Are you talking about matrix builds? Maybe we should use the term "matrix builds" itself to stay consistent with our messaging.

File: sources/pipelines_configure.md [hunk 6]
```diff
@@ -113,8 +113,53 @@ To create an environment on Shippable, you should have installed the Universal C
...
+  on_failure:
+
+integrations:
+  deploy:
+  hub:
+  notifications:
+  key:
+```
+* If `prov` section of `infra` tag is empty, default command `terraform apply` will be executed.
+* Your state files are saved at `/shippableci/shippable/provision` directory before provisioning, in terraform commands you can specify path to state files as shown in the above shippable.yml file.
```
Comment (rageshkrishna) [inline sources/pipelines_configure.md]: This line is unclear. I don't understand what you're trying to say here.

Comment (ankul-shippable) [inline sources/pipelines_configure.md]: @rageshkrishna this is to give information about path to state files to user.

**Comment by rageshkrishna** [comment]
LGTM. Tagging @abhijitkini and @manishas for a second look.

**Pull request merged**